### PR TITLE
Add support for routes

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

File: src/netlink.c [hunk 1]
```diff
@@ -404,44 +407,78 @@ process_ipv4_gw(const struct nlmsghdr *nlh, void *data)
...
  *
- * \return \c string value of the gateway,
- * or \c "" if there is no default gateway.
+ * \return true on success, false otherwise.
  */
-gchar *
-netlink_get_default_gw(struct netlink_handle *const hndl,
-		   guint8 family)
+gboolean
+netlink_get_routes(struct cc_oci_config *config,
```
Comment (jodh-intel) [inline src/netlink.c]: Need to check `config` param.

File: src/networking.c [hunk 2]
```diff
@@ -141,6 +141,22 @@ cc_oci_net_interface_free (struct cc_oci_net_if_cfg *if_cfg)
...
+ */
+void cc_oci_net_ipv4_route_free(struct cc_oci_net_ipv4_route *route)
+{
+	if (! route) {
+		return;
+	}
+
+	g_free_if_set (route->dest);
+	g_free_if_set (route->ifname);
+	g_free_if_set (route->gateway);
```
Comment (jodh-intel) [inline src/networking.c]: `route` itself isn't being freed here.

File: src/oci.h [hunk 3]
```diff
@@ -326,8 +323,22 @@ struct cc_oci_net_cfg {
 	GSList  *interfaces;
 
 	/** TODO: Add support for routes */
```
Comment (jodh-intel) [inline src/oci.h]: Could you remove this comment as it is now incorrect.

**Comment by jodh-intel** [comment]
@amshinde - thanks for updating the branch.

lgtm.

**Comment by dlespiau** [comment]
lgtm

**Comment by chavafg** [comment]
qa-failed

**Comment by sameo** [comment]
LGTM

**Comment by dlespiau** [comment]
lgtm

File: src/hypervisor.c [hunk 4]
```diff
@@ -56,12 +56,11 @@ cc_oci_expand_net_cmdline(struct cc_oci_config *config) {
 		return NULL;
 	}
 
-	if (! (config->net.gateway && config->net.hostname)) {
+	if (! config->net.hostname) {
 		return NULL;
 	}
 
-	return ( g_strdup_printf("ip=:::%s:::%s::off::",
```
Comment (jcvenegas) [inline src/hypervisor.c]: yay!

**Pull request merged**